hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieInputFormatUtils.java

```diff
@@ -527,19 +529,39 @@ public static String getTableBasePath(InputSplit split, JobConf jobConf) throws
       return realtimeSplit.getBasePath();
     } else {
       Path inputPath = ((FileSplit) split).getPath();
-      FileSystem fs = inputPath.getFileSystem(jobConf);
-      HoodieStorage storage = new HoodieHadoopStorage(fs);
-      Option<StoragePath> tablePath = TablePathUtils.getTablePath(storage, convertToStoragePath(inputPath));
-      return tablePath.get().toString();
+      return getTablePath(jobConf, inputPath);
     }
   }
 
+  private static String getTablePath(JobConf jobConf, Path inputPath) throws IOException {
+    FileSystem fs = inputPath.getFileSystem(jobConf);
+    HoodieStorage storage = new HoodieHadoopStorage(fs);
+    Option<StoragePath> tablePath = TablePathUtils.getTablePath(storage, convertToStoragePath(inputPath));
+    return tablePath.get().toString();
+  }
+
   /**
    * `schema.on.read` and skip merge not implemented
    */
-  public static boolean shouldUseFilegroupReader(final JobConf jobConf, final InputSplit split) {
-    return jobConf.getBoolean(HoodieReaderConfig.FILE_GROUP_READER_ENABLED.key(), HoodieReaderConfig.FILE_GROUP_READER_ENABLED.defaultValue())
-        && !jobConf.getBoolean(HoodieCommonConfig.SCHEMA_EVOLUTION_ENABLE.key(), HoodieCommonConfig.SCHEMA_EVOLUTION_ENABLE.defaultValue())
-        && !(split instanceof BootstrapBaseFileSplit);
+  public static boolean shouldUseFilegroupReader(final JobConf jobConf, final InputSplit split) throws IOException {
+    if (split instanceof FileSplit || split instanceof RealtimeSplit) {
+      HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(getStorageConf(jobConf)).setBasePath(getTableBasePath(split, jobConf)).build();
```

Comment:
lets confirm to get this reviewed from @jonvex

Reply:
can you file a follow up jira to get this reviewed by ethan. for now, we can go ahead and land this patch.

Reply:
Let's add a JIRA ticket and tag the fix version `1.0.2`.

Reply:
we will file a ticket for review once we have the follow up patch.